src/storage.h

```diff
@@ -0,0 +1,260 @@
+#ifndef CYCLUS_STORAGES_STORAGE_H_
+#define CYCLUS_STORAGES_STORAGE_H_
+
+#include <string>
+
+#include "cyclus.h"
+
+// forward declaration
+namespace storage {
+class Storage;
+} // namespace storage
+
+
+namespace storage {
+/// @class Storage
+///
+/// This Facility is intended to hold materials for a user specified
+/// amount of time in order to model a storage facility with a certain
+/// process time or holdup time.
+/// The Storage class inherits from the Facility class and is
+/// dynamically loaded by the Agent class when requested.
+///
+/// @section intro Introduction
+/// This Agent was initially developed to support the fco code-to-code 
+/// comparison.
+/// It's very similar to the "NullFacility" of years 
+/// past. Its purpose is to hold materials and release them only  
+/// after some period of delay time.
+///
+/// @section agentparams Agent Parameters
+/// in_commod is a string naming the commodity that this facility receives
+/// out_commod is a string naming the commodity that in_commod is stocks into
+/// process_time is the number of timesteps between receiving and offering
+/// in_recipe (optional) describes the incoming resource by recipe
+/// 
+/// @section optionalparams Optional Parameters
+/// max_inv_size is the maximum capacity of the inventory storage
+/// capacity is the maximum processing capacity per timestep
```

Comment:
We've struggled with the fact that "capacity" is a term that can be ambiguous, referring to both the static capacity of an inventory and also the throughput of a facility.

We have tended towards the variable `throughput` to be less ambiguous.